src/main/webapp/app/overview/exercise-details/exercise-details-student-actions.component.ts

```
@@ -386,12 +388,27 @@
 
     hasAthenaResultForlatestSubmission(): boolean {
         if (this.gradedParticipation?.submissions && this.gradedParticipation?.results) {
-            // submissions.results is always undefined so this is neccessary
-            return (
-                this.gradedParticipation.submissions.last()?.id ===
-                this.gradedParticipation?.results.filter((result) => result.assessmentType == AssessmentType.AUTOMATIC_ATHENA).first()?.submission?.id
-            );
+            const sortedSubmissions = this.gradedParticipation.submissions.slice().sort((a, b) => {
+                const dateA = this.getDateValue(a.submissionDate) ?? -Infinity;
+                const dateB = this.getDateValue(b.submissionDate) ?? -Infinity;
```

Biome

[error] Use Number.Infinity instead of the equivalent global.

ES2015 moved some globals into the Number namespace for consistency.
Safe fix: Use Number.Infinity instead.


(lint/style/useNumberNamespace)